server/container/guice/blob-export-guice/src/main/java/org/apache/james/modules/BlobExportImplChoice.java

```diff
@@ -0,0 +1,112 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.modules;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.apache.commons.configuration.Configuration;
+import org.apache.commons.configuration.ConfigurationException;
+
+import com.github.steveash.guavate.Guavate;
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+class BlobExportImplChoice {
+
+    enum BlobExportImplName {
+        LOCAL_FILE("localFile");
+
+        private static Optional<BlobExportImplName> from(String implNameString) {
+            Preconditions.checkNotNull(implNameString);
+
+            return Stream.of(values())
+                .filter(impl -> impl.name.equals(implNameString))
+                .findFirst();
+        }
+
+        private static ImmutableList<String> plainImplNames() {
+            return Stream.of(values())
+                .map(impl -> impl.name)
+                .collect(Guavate.toImmutableList());
+        }
+
+        private final String name;
+
+        BlobExportImplName(String name) {
+            this.name = name;
+        }
+
+        String getImplName() {
+            return name;
+        }
+    }
+
+    static BlobExportImplChoice localFile() {
+        return new BlobExportImplChoice(BlobExportImplName.LOCAL_FILE);
+    }
+
+    static BlobExportImplChoice from(Configuration configuration) throws ConfigurationException {
+        String blobExportImpl = configuration.getString(BLOB_EXPORT_MECHANISM_IMPL);
+
+        String sanitizedImplName = Optional.ofNullable(blobExportImpl)
+            .map(String::trim)
+            .orElseThrow(() -> new ConfigurationException(BLOB_EXPORT_MECHANISM_IMPL + " property is mandatory"));
```

Comment:
In that case we should supply a viable alternative: localFile

Reply:
my work follows the impl of `BlobStoreChoosingConfiguration` where it force users to specify blobStore name, otherwise it throws. I did the same thing with BlobExport.
If returning the default value is the choice, then I see the in consistent in the way we handle properites

Reply:
We should avoid breaking changes as much as possible.

So, when I do an upgrade, I should as an user not be required to change my configuration.

We should do a log warning + use safe defaults.